### PR TITLE
Request basic crud

### PR DESCRIPTION
**Pull request by OleksiiPliekhov**

### CODE REVIEW

File: src/main/java/com/shinobicoders/teamcodeapi/model/Request.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        return Objects.hashCode(this.id);
+    }
+
+    @Override
+    public boolean equals(Object o){
+        if(this == o) return true;
+        if(o == null || o.getClass() != this.getClass()) return  false;
+
+        Request request = (Request) o;
+        return Objects.equals(this.id, request.getId()) && this.status == request.getStatus()
```
Comment (havriutkin) [inline src/main/java/com/shinobicoders/teamcodeapi/model/Request.java]: It is enough to compare userID and ProjectID. Maybe we should change DB and have (userId, ProjectId) as PK

File: src/main/java/com/shinobicoders/teamcodeapi/service/RequestService.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        return (List<Request>) requestRepository.findAll();
+    }
+
+    public Request getRequestById(Long id){
+        return requestRepository.findById(id).orElseThrow(() -> new EntityNotFoundException("Request with ID [" + id + "] not found"));
+    }
+
+    public Request createRequest(Request request){
+        List<Request> requests = getAllRequests();
+        Optional<Request> existingRequest = requests.stream().filter(i -> i.getProject().equals(request.getProject())
```
Comment (havriutkin) [inline src/main/java/com/shinobicoders/teamcodeapi/service/RequestService.java]: After changing equals() method in Request class you can make it easier

**Review by havriutkin — CHANGES_REQUESTED** [review]
We should reconsider PK in Request table

File: src/main/java/com/shinobicoders/teamcodeapi/model/Request.java (hunk 1 above)
Comment (OleksiiPliekhov) [reply src/main/java/com/shinobicoders/teamcodeapi/model/Request.java]: Maybe we just should create a constraint of uniqueness Project-User pair in Request

File: src/main/java/com/shinobicoders/teamcodeapi/model/Request.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+        return Objects.hashCode(this.id);
+    }
+
+    @Override
+    public boolean equals(Object o){
+        if(this == o) return true;
+        if(o == null || o.getClass() != this.getClass()) return  false;
+
+        Request request = (Request) o;
+        return this.getUser().equals(request.getUser())
```
Comment (havriutkin) [inline src/main/java/com/shinobicoders/teamcodeapi/model/Request.java]: Compare only Id

File: src/main/java/com/shinobicoders/teamcodeapi/model/Request.java [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    private RequestStatus status;
+
+    private Date requestDate;
+
+    private String message;
+
+    @ManyToOne
```
Comment (havriutkin) [inline src/main/java/com/shinobicoders/teamcodeapi/model/Request.java:22]: Add unique constraint on {user, project}

File: src/main/java/com/shinobicoders/teamcodeapi/service/RequestService.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+            return requestRepository.save(request);
+        }
+    }
+
+    public Request updateRequestMessage(Long id, String newMessage){
+        Request request = requestRepository.findById(id).orElseThrow(() -> new EntityNotFoundException("Request with ID [" + id + "] not found"));
+        request.setMessage(newMessage);
+        return request;
+    }
+
```
Comment (havriutkin) [inline src/main/java/com/shinobicoders/teamcodeapi/service/RequestService.java:45]: Add updateRequestStatus

**Review by havriutkin — CHANGES_REQUESTED** [review]

**Pull request merged**